### PR TITLE
DVCSMP-2097 - alterSetpoint - check for auto or off mode

### PR DESCRIPTION
**Pull request by james-smartthings**
In alterSetpoint function, added a check to check for "auto" or "off" mode. This fixes the issues in JIRA Ticket DVCSMP-2097

### CODE REVIEW

File: devicetypes/smartthings/ecobee-thermostat.src/ecobee-thermostat.groovy [hunk 1]
```diff
@@ -655,55 +655,60 @@ void lowerSetpoint() {
...
 
 	def mode = device.currentValue("thermostatMode")
-	def heatingSetpoint = device.currentValue("heatingSetpoint")
-	def coolingSetpoint = device.currentValue("coolingSetpoint")
-	def deviceId = device.deviceNetworkId.split(/\./).last()
 
-	def targetHeatingSetpoint
-	def targetCoolingSetpoint
+	if (mode == "off" || mode == "auto") {
+		log.warn "this mode: $mode does not allow lowerSetpoint"
```
Comment (dsainteclaire) [inline devicetypes/smartthings/ecobee-thermostat.src/ecobee-thermostat.groovy]: consider changing "lowerSetpoint" to "alterSetpoint"

**Comment by dsainteclaire** [comment]
Just the one comment.  Other-than that, I'd test it with different modes and stuff before you merge it and it would be good to merge

File: devicetypes/smartthings/ecobee-thermostat.src/ecobee-thermostat.groovy (hunk 1 above)
Comment (james-smartthings) [reply devicetypes/smartthings/ecobee-thermostat.src/ecobee-thermostat.groovy]: log has been changed to "alterSetpoint"

**Comment by james-smartthings** [comment]
Ecobee functionality is working correctly with this modification.

**Comment by dsainteclaire** [comment]
LGTM ⛩

**Comment by workingmonk** [comment]
please add the ticket number to the PR title

**Pull request merged**